### ISSUE TITLE
Change one-var to support TSLint's "ignore-for-loop"

### ISSUE THREAD

**Issue by gruckion** [eslint/eslint#11786]
**What rule do you want to change?**

https://eslint.org/docs/rules/one-var to support TSLint options "ignore-for-loop"
https://palantir.github.io/tslint/rules/one-variable-per-declaration

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

new option

**Please provide some example code that this change will affect:**

Allowed if option `"ignore-for-loop": true`
```js
for (var i = 0, j = 1, n = 2; i < 50, n < 50; i = i + 3, j = j + 3, n = n + 3){
    console.log("variable i: " + i);
    console.log("variable j: " + j);
    console.log("variable n: " + n);
}
```
Still warns on
```js
/*eslint one-var: ["error", "always"]*/
/*eslint-env es6*/

function foo() {
    var bar,
        baz;
    let qux,
        norf;
}
```

Example config

```json
        "one-var": [
            "error",
            {
                "var": "always-in-loop",
                "let": "never",
                "const": "never-in-loop"
            }
        ],
```

**What does the rule currently do for this code?**

Warns of errors on the use of comma separated variable declarations inside a for loop.

**What will the rule do after it's changed?**

Optionally not warn/error for comma separated variable declarations inside a for loop

**Are you willing to submit a pull request to implement this change?**

Maybe

**Comment by g-plane** [comment]
Hey @gruckion , thanks for the issue. I like this proposal, but we need 2 more upvotes from the team to support it.

**Comment by mysticatea** [comment]
I wonder if it's the current behavior: [online demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgb25lLXZhcjogW2Vycm9yLCBuZXZlcl0gKi9cclxuXHJcbmZ1bmN0aW9uIGZvbygpIHtcclxuICAgIHZhciBiYXIsXHJcbiAgICAgICAgYmF6O1xyXG4gICAgbGV0IHF1eCxcclxuICAgICAgICBub3JmO1xyXG59XHJcblxyXG5mb3IgKHZhciBpID0gMCwgaiA9IDEsIG4gPSAyOyBpIDwgNTAsIG4gPCA1MDsgaSA9IGkgKyAzLCBqID0gaiArIDMsIG4gPSBuICsgMyl7XHJcbiAgICBjb25zb2xlLmxvZyhcInZhcmlhYmxlIGk6IFwiICsgaSk7XHJcbiAgICBjb25zb2xlLmxvZyhcInZhcmlhYmxlIGo6IFwiICsgaik7XHJcbiAgICBjb25zb2xlLmxvZyhcInZhcmlhYmxlIG46IFwiICsgbik7XHJcbn1cclxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19).

**Comment by gruckion** [comment]
When I tested this it did not appear to work. I can't recall the exact code I had an issue with.

@mysticatea I agree, the above proposed examples are covered in the current behavior, hence this should be closed.

Please close if you agree.

**Comment by mysticatea** [comment]
OK. Please open a new issue if you find buggy behavior.

Thank you.

**Comment by JoshuaKGoldberg** [comment]
@mysticatea could you please reopen this issue? TSLint by default _does_ check for loops, while ESLint has no way to configure this. An option like `{ "checkForLoops": true }` would be helpful in maintaining feature parity when moving from TSLint to ESLint _(and, in general, a nice to have!)_.

**Comment by kaicataldo** [comment]
Please feel free to create a new issue with your proposal. Thanks!
